### DISCUSSION TITLE
Is there a command/API to allow me refresh neo-tree after `vim_buffer_changed` event?

### DISCUSSION THREAD

**Discussion by linrongbin16** [nvim-neo-tree/neo-tree.nvim#1234] [Q&A] [answered]
Hi, I found that after I delete/add/change vim buffer, the neo-tree won't auto-update, I want to use `vim_buffer_changed` event to auto-update, is there a command to do it?

For now I'm doing it manually, e.g., just switch to neo-tree on the left side, and press `R` to refresh it.

**Comment by cseickel** [comment]
It should already be doing that. Can you tell me what source you are talking about (filesystem?) and share your config?

**Reply by linrongbin16** [reply]
yes, it's filesystem, my config is here: https://github.com/linrongbin16/lin.nvim/blob/main/lua/configs/nvim-neo-tree/neo-tree-nvim/config.lua#L19.

**Reply by cseickel** [reply]
I'm confused, in the filesystem, what would an update do with regards to add/change/delete buffers?

**Reply by linrongbin16** [reply]
emmm, usually some changes would affect  git status and lsp diagnostics status.

**Reply by cseickel** [reply]
That should all work fine but I confirmed it does not with your config. It's a bug and I'll look at it tonight.

**Reply by linrongbin16** [reply]
thanks!

**Comment by cseickel** [comment, marked as answer]
This should be fixed in main: https://github.com/nvim-neo-tree/neo-tree.nvim/pull/1237

It will be released tonight or tomorrow.

**Reply by linrongbin16** [reply]
thank you!
